blocks/init/src/Blocks/custom/featured-posts/components/featured-posts-options.js

```diff
@@ -84,28 +153,70 @@ export const FeaturedPostsOptions = ({ attributes, setAttributes }) => {
 				<Spinner />
 			}
 
-			{(postTypeOptions[0] && posts) ?
+			{(postTypeOptions && taxonomyOptions) ?
 				<SelectControl
-					label={__('Posts Items', 'eightshift-frontend-libs')}
-					value={posts}
-					multiple
-					options={postsOptions}
+					label={sprintf(__('Filter by %s available taxonomies', 'eightshift-frontend-libs'), _.startCase(_.toLower(postType)))}
+					help={sprintf(__('If `No Filter` value is selected your %s posts will not be filtered by any category.', 'eightshift-frontend-libs'), _.startCase(_.toLower(postType)))}
+					value={taxonomy}
+					options={taxonomyOptions}
 					onChange={(value) => {
 						setAttributes({
 							query: {
 								...queryProps,
-								posts: value[0] ? value : [],
+								taxonomy: value,
+								terms: [],
+								posts: [],
 							},
 						});
 					}}
 				/> :
 				<Spinner />
 			}
 
+			{(taxonomyOptions && taxonomy) &&
+				<CustomSelect
+					label={sprintf(__('Filter by %s taxonomy', 'Tvornica'), _.startCase(_.toLower(taxonomy)))}
```

Comment:
The same comment with extracting out the `_.startCase(_.toLower(taxonomy))` part.